specification/cdn/resource-manager/Microsoft.Cdn/stable/2021-06-01/afdx.json

```diff
@@ -5289,6 +5355,65 @@
         }
       }
     },
+    "CheckAfdEndpointNameAvailabilityInput": {
+      "description": "Input of CheckNameAvailability API.",
+      "type": "object",
+      "required": [
+        "name",
+        "type"
+      ],
+      "properties": {
+        "name": {
+          "description": "The resource name to validate.",
+          "type": "string"
+        },
+        "type": {
+          "description": "The type of the resource whose name is to be validated.",
+          "$ref": "#/definitions/AfdResourceType"
+        },
+        "autoGeneratedDomainNameLabelScope": {
+          "description": "Indicates the endpoint name reuse scope. The default value is TenantReuse.",
+          "$ref": "#/definitions/AutoGeneratedDomainNameLabelScope"
+        }
+      }
+    },
+    "CheckAfdEndpointNameAvailabilityOutput": {
+      "description": "Output of check name availability API.",
+      "type": "object",
+      "properties": {
+        "nameAvailable": {
+          "description": "Indicates whether the name is available.",
+          "readOnly": true,
+          "type": "boolean"
+        },
+        "availableHostname": {
+          "description": "Returns the available hostname generated based on the AutoGeneratedDomainNameLabelScope when the name is available, otherwise it returns empty string",
+          "readOnly": true,
+          "type": "string"
+        },
+        "reason": {
+          "description": "The reason why the name is not available.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "message": {
+          "description": "The detailed error message describing why the name is not available.",
+          "readOnly": true,
+          "type": "string"
+        }
+      }
+    },
+    "AfdResourceType": {
```

Comment:
This shoudln;t be required. We can reuse ResourceType from cdn.json

Reply:
Fixed.